src/plugins/discord/fetcher.test.js

```diff
@@ -0,0 +1,299 @@
+// @flow
+
+import {Fetcher} from "./fetcher";
+import {
+  type Channel,
+  type Guild,
+  type Reaction,
+  type Message,
+  type GuildMember,
+  type Snowflake,
+} from "./models";
+
+/**
+ * Note: the 'any' type signature is assigned to server response objects
+ * to make explicit that they are untyped prior to transformation by the
+ * appropriate handlers
+ */
+
+describe("plugins/discord/fetcher", () => {
+  const defaultOptions = () => ({
+    membersLimit: 100,
+    messagesLimit: 100,
+    reactionsLimit: 100,
+  });
+
+  describe("fetch guild", () => {
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      await fetcher.guild("1");
+      expect(fetch.mock.calls[0]).toEqual(["guilds/1"]);
+    });
+
+    it("handles response", async () => {
+      const expected: Guild = {id: "1", name: "guildname", permissions: 0};
+      const fetch = jest.fn(() => Promise.resolve(expected));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      const guild = await fetcher.guild("1");
+      expect(guild).toEqual(expected);
+    });
+  });
+
+  describe("fetch channels", () => {
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      await fetcher.channels("1");
+      expect(fetch.mock.calls[0]).toEqual(["/guilds/1/channels"]);
+    });
+
+    it("handles response", async () => {
+      // response object from server, prior to type transformation
+      const testChannel: any = (id: string) => ({
```

Comment:
There's an issue with this type pattern.
Originally this was:
```
const testChannel = (id: string) => //...
```
Will be typed as `const testChannel` is a function with `(id: string)` as args and an implied `any` return type.
Now this would be cast from a Function to an `any`. This is less strict than it should be.

The intention was to make the **return type** an explicit `any`. With a comment to mention, because we're mocking a raw JSON server response (not a mapped model) the type is intentionally `any`.

So the pattern should be changed like:
```diff
- // Turns a function into any
- const testChannel: any = (id: string) => // ...

+ // Sets an explicit any return type
+ const testChannel = (id: string): any => // ...
```

Reply:
Updated, good catch, this was an oversight.